compiler-core/src/type_/lsp_pretty.rs

```diff
@@ -0,0 +1,93 @@
+use std::sync::Arc;
+
+use ecow::EcoString;
+use im::HashMap;
+use itertools::Itertools;
+
+
+use crate::ast::TypeAst;
+
+use super::{pretty::Printer, Type, TypeVar};
+
+#[derive(Debug)]
+pub struct LspPrinter<'a, > {
+    pretty_printer: Printer,
+    type_qualifiers: &'a HashMap<EcoString, EcoString>,
+    module_qualifiers: &'a HashMap<EcoString, EcoString>,
+    type_parameters: &'a HashMap<u64, TypeAst>,
+}
+
+impl<'a> LspPrinter<'a> {
+    pub fn new(
+        type_qualifiers: &'a HashMap<EcoString, EcoString>,
+        module_qualifiers: &'a HashMap<EcoString, EcoString>,
+        type_parameters: &'a HashMap<u64, TypeAst>,
+    ) -> Self {
+        LspPrinter {
+            pretty_printer: Printer::default(),
+            type_qualifiers,
+            module_qualifiers,
+            type_parameters
+        }
+    }
+
+    /// Render a Type as a well formatted string.
+    pub fn pretty_print(&mut self, typ: &Type) -> String {
+        self.print(typ)
+    }
+
+    pub fn print(&mut self, typ: &Type) -> String {
+        match typ {
+            Type::Named {
+                name, args, module, ..
+            } => {
+                let key = module.clone() + "." + name.clone();
+                let res = self.type_qualifiers.get(&key).unwrap_or(&key).to_string();
+                if args.is_empty() {
+                    if module == "gleam" {
+                        // Ignoring module names for in-built gleam types like Int
+                        name.to_string()
```

Comment:
This could produce incorrect results when the prelude types have been shadowed locally.